CONTRIBUTING.md

```diff
@@ -0,0 +1,116 @@
+    - Describe the environment that (re)produced the problem.
+
+> For a bug to be actionable, it needs to be reproducible. If you or
+> contributors can’t reproduce the bug, try to figure out why. Please take
+> care to stay involved in discussions around solving the problem.
+
+## Triaging bugs or contributing code
+
+If you're triaging a bug, try to reduce it. Once a bug can be reproduced,
+reduce it to the smallest amount of code possible. Reasoning about a sample
+or unit test that reproduces a bug in just a few lines of code is easier than
+reasoning about a longer sample.
+
+From a practical perspective, contributions are as simple as:
+- Forking the repository on GitHub.
+- Making changes to your forked repository.
+- When committing, reference your issue (if present) and include a note about
+  the fix.
+- If possible, and if applicable, please also add/update unit tests for your
+  changes.
+- Push the changes to your fork and submit a pull request to the 'master'
+  branch of the projects' repository.
+
+If you are interested in making a large change and feel unsure about its
+overall effect, please make sure to first discuss the change and reach a
+consensus with core contributors through [slack]. Then ask about the best way
+to go about making the change.
+
+## Code Review
+
+Any open source project relies heavily on code review to improve software
+quality:
+
+> All significant changes, by all developers, must be reviewed before they
+> are committed to the repository. Code reviews are conducted on GitHub through
+> comments on pull requests or commits. The developer responsible for a code
+> change is also responsible for making all necessary review-related changes.
+
+Sometimes code reviews will take longer than you would hope for, especially
+for larger features. Here are some accepted ways to speed up review times for
+your patches:
+
+- Review other people’s changes. If you help out, everybody will be more
+  willing to do the same for you. Goodwill is our currency.
+- Split your change into multiple smaller changes. The smaller your change,
+  the higher the probability that somebody will take a quick look at it.
+- Ping the change on [slack]. If it is urgent, provide reasons why it is
+  important to get this change landed. Remember that you’re asking for
+  valuable time from other professional developers.
+
+**Note that anyone is welcome to review and give feedback on a change, but
+only people with commit access to the repository can approve it.**
+
+## Attribution of Changes
+
+When contributors submit a change to this project, after that change is
+approved, other developers with commit access may commit it for the author.
+When doing so, it is important to retain correct attribution of the
+contribution. Generally speaking, Git handles attribution automatically.
+
+## Code Documentation
+
+Ensure that every function in `warehouse.ai` is documented and follows the
```

Comment:
>  `warehouse.ai`
Should that be `extract-config`?

Reply:
Yes, will change it so that this becomes repo agnostic, e.g. `this module` or `this project`